youtube_dl/YoutubeDL.py

```diff
@@ -1819,7 +1827,14 @@ def ensure_dir_exists(path):
                     self.report_error('Cannot write metadata to JSON file ' + infofn)
                     return
 
-        self._write_thumbnails(info_dict, filename)
+        if self.params.get('writethumbnail', False):
+            self._write_thumbnails(info_dict, filename)
+        if self.params.get('write_all_thumbnails', False):
+            self._write_thumbnails(info_dict, filename, write_all=True)
+        if self.params.get('writestoryboard', False):
+            self._write_thumbnails(info_dict, filename, param_name='storyboards')
+        if self.params.get('write_all_storyboards', False):
+            self._write_thumbnails(info_dict, filename, param_name='storyboards', write_all=True)
```

Comment:
Code duplication. Keep the original `_write_thumbnails` code and generalize item part. `param_name` is a bad name for something that describe item kind.

Reply:
I'm sorry, but I don't get what do you mean here. 
If I always call _write_thumbnails for both items (without checking self.params here) and do all the self.params checks inside the _write_thumbnails function, then I'm just moving around the same ifs somewhere else, aren't I?